src/main/java/it/pagopa/pn/bff/service/InfoPaService.java

```diff
@@ -0,0 +1,55 @@
+package it.pagopa.pn.bff.service;
+
+import it.pagopa.pn.bff.exceptions.PnBffException;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffInstitution;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffInstitutionProduct;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.CxTypeAuthFleet;
+import it.pagopa.pn.bff.mappers.InfoPaMapper;
+import it.pagopa.pn.bff.pnclient.externalregistries.PnExternalRegistriesClientImpl;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class InfoPaService {
+    private final PnExternalRegistriesClientImpl pnExternalRegistriesClient;
+
+    @Value("${pn.selfcare.baseurl}")
+    private String selfcareUrl;
+
+    @Value("${pn.selfcare.send.prod.id}")
+    private String prodId;
+
+    public Flux<BffInstitution> getInstitutions(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType, String xPagopaPnCxId, String xPagopaPnSrcCh, List<String> xPagopaPnCxGroups, String xPagopaPnSrcChDetails) {
+        log.info("getInstitutions");
+        String pathTokenExchange = "/token-exchange?institutionId=";
+        String pathProdId = "&productId=" + prodId;
+        return pnExternalRegistriesClient
```

Comment:
Do this transformations into the mapper

Reply:
This can't be done because it's not possible to retrieve an existing instance (thus using `@Autowired`) within an interface.
Additionally, this involves business logic and should therefore be integrated into the service, as we're manipulating data by concatenating multiple strings. This isn't a basic mapping model where entities are simply mapped to DTOs without modification